### PR TITLE
fix: migrate entry injection to hooks

### PR DESCRIPTION
**Pull request by rishabh3112**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
 Shouldn't require change 

### Motivation / Use-Case
/cc @alexander-akait 
Helps fixing #3351 

### Breaking Changes
No

### Additional Info
No

### CODE REVIEW

**Comment by snitin315** [comment]
Need to rebase.

**Comment by rishabh3112** [comment]
Need review 😃

File: lib/utils/DevServerPlugin.js [hunk 1]
```diff
@@ -252,12 +252,49 @@ class DevServerPlugin {
...
     // use a hook to add entries if available
     if (typeof webpack.EntryPlugin !== 'undefined') {
-      for (const additionalEntry of additionalEntries) {
-        new webpack.EntryPlugin(compiler.context, additionalEntry, {
-          // eslint-disable-next-line no-undefined
-          name: undefined,
-        }).apply(compiler);
-      }
+      // Register Entry Dependency in Factory
+      const EntryDependency = require('webpack/lib/dependencies/EntryDependency');
```
Comment (alexander-akait) [inline lib/utils/DevServerPlugin.js:256]: Avoid using this, webpack should public export this

File: lib/utils/DevServerPlugin.js [hunk 2]
```diff
@@ -252,12 +252,49 @@ class DevServerPlugin {
...
+      compiler.hooks.compilation.tap(
+        'webpack-dev-server',
+        (compilation, { normalModuleFactory }) => {
+          compilation.dependencyFactories.set(
+            EntryDependency,
+            normalModuleFactory
+          );
+        }
+      );
```
Comment (alexander-akait) [inline lib/utils/DevServerPlugin.js:265]: Webpack already have this factory

File: lib/utils/DevServerPlugin.js (hunk 1 above)
Comment (rishabh3112) [reply lib/utils/DevServerPlugin.js:256]: Should I send PR to webpack then?

Comment (rishabh3112) [reply lib/utils/DevServerPlugin.js:254]: @alexander-akait 😅

**Comment by ylemkimon** [comment]
Note this is no longer needed as of #3467.

**Comment by rishabh3112** [comment]
Closing this PR.
@alexander-akait  if this is needed, let me know. I would start fresh.

**Pull request closed without merging**